src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -36,6 +37,37 @@ public class PackageArchiveReader : PackageReaderBase
         /// </summary>
         protected Stream ZipReadStream { get; set; }
 
+        /// <summary>
+        /// Nupkg package reader
+        /// </summary>
+        /// <param name="frameworkProvider">Framework mapping provider for NuGetFramework parsing.</param>
+        /// <param name="compatibilityProvider">Framework compatibility provider.</param>
+        private PackageArchiveReader(IFrameworkNameProvider frameworkProvider, IFrameworkCompatibilityProvider compatibilityProvider)
+            : base(frameworkProvider, compatibilityProvider)
+        {
+            _environmentVariableReader = EnvironmentVariableWrapper.Instance;
+        }
+
+        // For testing purposes only
+        internal PackageArchiveReader(Stream stream, IEnvironmentVariableReader environmentVariableReader)
+            : this(stream)
+        {
+            if (environmentVariableReader != null)
```

Comment:
Should we have a null check instead here? 

Otherwise there's a world where this could NPE.

Reply:
No, because this one is coming from unit test. Actual value is already set in base constructor in line 48, so it'll be never be null under.

Reply:
This one only ensures we're not replacing good value with bad value for unit tests.